R/db.R

```diff
@@ -202,42 +202,29 @@ as_wkb <- function(x) {
 
 get_possibly_new_srid <- function(conn, crs) {
 
-	# Return if we have a matching srid
 	db_crs <- find_database_srid(conn, crs)
-	if(is_crs(db_crs) & !is.na(db_crs)) {
+	if(!is.na(db_crs)) {
 		return(db_crs)
 	}
 
-	# No matching SRID, so compare wkt if available
-
-    srs_table = try(dbReadTable(conn, "spatial_ref_sys"))
-
-    if (class(srs_table) == "try-error") {  # nocov start
-        warning("Could not find table `spatial_ref_sys` on remote connexion; ",
-                "CRS is set to unknown.")
-        return(0)
-    } # nocov end
+	db_crs <- find_database_srtext(conn, crs)
+	if (!is.na(db_crs)) {
+		return(db_crs)
+	}
 
-    srs_table$proj4text = sapply(srs_table$proj4text, trim)
-    eq = srs_table$proj4text == proj4string
-    if (any(eq))
-        srid <- srs_table[min(which(eq)) , "srid"]
-    else { # create a new srid in conn if proj4string is not found:
-        srid <- get_new_postgis_srid(conn)
-        set_postgis_crs(conn, st_crs(srid, proj4string, valid = FALSE))
-    }
-    srid
+	db_insert_crs(conn, crs)
 }
+
 #' Find srid in a database by using the srid
```

Comment:
So you're writing documentation for a function that is not exported, right? If so, why? I'd suggest to leave out the `#'`, maybe use only `#`, so that no documentation objects are created.

Reply:
Yeah, it feels awkward to write a .Rd, but I still felt like minimal documentation would be useful. I'll use `#`, then.